addons/l10n_ch/models/res_bank.py

```diff
@@ -104,6 +138,33 @@ def _compute_sanitized_acc_number(self):
             bank.sanitized_acc_number = bank.acc_number
         super(ResPartnerBank, self - postal_banks)._compute_sanitized_acc_number()
 
+    @api.depends('acc_number')
+    def _compute_l10n_ch_qr_iban(self):
+        for record in self:
+            try:
+                validate_qr_iban(record.acc_number)
+                valid_qr_iban = True
+            except ValidationError:
+                valid_qr_iban = False
+
+            if valid_qr_iban:
+                record.l10n_ch_qr_iban = record.sanitized_acc_number
+            else:
+                record.l10n_ch_qr_iban = None
```

Comment:
@oco-odoo Does this mean it won't be changed if it already had a value?  (None instead of "")

Reply:
@jco-odoo But if we do that, if the users enters a wrong acc number which is a QR-IBAN, then fixes the acc number with another QR-IBAN, the QR-IBAN field from the wrong account will be kept. Whatever the choice, there will always be limitations; I'd say it's better to keep things simple. It's already like this in a lot of places within Odoo. 
Plus, I think 99.99% of the people will enter the acc_number first anyway.

Reply:
Makes sense.